flips/20211007-transaction-fees.md

```diff
@@ -0,0 +1,327 @@
+
+## Design requirements
+
+The requirements for transaction fees can be separated into two groups depending on perspective.
+
+From the protocol's perspective the following requirements can be set:
+
+- Fees should be proportional to effort: Transactions that require more resources (cpu, memory, bandwidth, storage, ...) from the network, to be processed, should cost more.
+- Surge pricing: If the protocol is experiencing a lot of traffic, all transactions should become more expensive, until the high traffic subsides.
+
+The ultimate goal from the protocols perspective is that transactions cost exactly the amount of FLOW needed to pay for the cost of the resources needed by the nodes to process the transaction. However measuring the exact resource consumption would be difficult and costly in itself, instead the goal is to make a reasonably good approximation and improve that approximation over time.
+
+From the users perspective the implementation of transaction fees should satisfy the following criteria:
+
+- Fees should be easy to understand:
+    - It should be made clear why the transaction fees exist in the form they do.
+    - It should be easy to explain how the fees are calculated.
+- It should be possible to calculate the minimum and maximum boundaries of the transaction fees for a transaction (within a reasonable margin), before actually sending a transaction.
+- The price of sending a specific transaction should not fluctuate over time so quickly that the user cannot respond to the fluctuation.
+- It should be possible to see the details of the transaction fee calculation after the transaction is executed.
+
+## Proposed design
+
+The idea behind this design is to break down the transaction fees into smaller parts that are easier to define, control, and implement.
+
+The motivation behind the first separation is to separate the fees into a part that can be known before executing the transaction script, inclusion fees, and a part that can only be know by executing the transaction script, execution fees. 
+
+The inclusion part of fees accounts for the resources needed by a transaction because of the transaction's properties (that can be known without executing the transaction's script). Transaction properties are things like the byte size of the transaction and the number of signatures. Some examples of resources that will be needed are the bandwidth to transfer the transaction from node to node, the verification of signatures and the verification of the sequence number.
+
+
+The execution part of fees accounts for the approximate operational cost of executing the transaction script, passing the results to the verification node and verifying the results. Execution fees cannot be know without actually executing the transaction script. In order to avoid having unexpectedly high execution fees, each transaction should specify a stopping limit for the execution fees.
```

Comment:
```suggestion
The execution part of fees accounts for the approximate operational cost of executing the transaction script, passing the results to the verification node and verifying the results. Execution fees cannot be known without actually executing the transaction script. In order to avoid having unexpectedly high execution fees, each transaction should specify a stopping limit for the execution fees.
```